src/core/control/jobs/RenderJob.cpp

```diff
@@ -52,6 +52,8 @@ void RenderJob::rerenderRectangle(Rectangle<double> const& rect) {
     cairo_set_source_surface(crPageBuffer.get(), rectBuffer.get(), 0, 0);
     cairo_rectangle(crPageBuffer.get(), rect.x, rect.y, rect.width, rect.height);
     cairo_fill(crPageBuffer.get());
+
+    this->view->repaintArea(rect.x, rect.y, rect.x + rect.width, rect.y + rect.height);
```

Comment:
nit: I think the code is more readable if this line is moved into the loop where the function is being called, so that all of the `view->repaint`-type calls will be in the `run()` function.